conda/common/io.py

```diff
@@ -31,7 +31,40 @@
 
 log = getLogger(__name__)
 
-_FORMATTER = Formatter("%(levelname)s %(name)s:%(funcName)s(%(lineno)d): %(message)s")
+
+class DeltaSecondsFormatter(Formatter):
+    """
+    Logging formatter with additional attributes for run time logging.
+
+    Attributes:
+      `delta_secs`:
+        Elapsed seconds since last log/format call (or creation of logger).
+      `relative_created_secs`:
+        Like `relativeCreated`, time relative to the initialization of the
+        `logging` module but conveniently scaled to seconds as a `float` value.
+    """
+    def __init__(self, fmt=None, datefmt=None):
+        self.prev_time = time()
+        super(DeltaSecondsFormatter, self).__init__(fmt=fmt, datefmt=datefmt)
+
+    def format(self, record):
+        now = time()
+        prev_time = self.prev_time
+        self.prev_time = max(self.prev_time, now)
+        record.delta_secs = now - prev_time
+        record.relative_created_secs = record.relativeCreated / 1000
+        return super(DeltaSecondsFormatter, self).format(record)
+
+
+if os.environ.get('CONDA_INSTRUMENTATION_ENABLED'):
```

Comment:
I was going to say I was fine with it, and then I looked at the actual output.

```
 2284.186     0.001 DEBUG conda.core.subdir_data:_load(216): Local cache timed out for https://repo.anaconda.com/pkgs/main/linux-64 at /home/circleci/.conda/pkgs/cache/47929eba.json
 2284.188     0.002 DEBUG conda.core.subdir_data:_read_pickled(287): found pickle file /home/circleci/.conda/pkgs/cache/98cb6fa3.q
 2284.190     0.004 DEBUG conda.core.subdir_data:_read_pickled(287): found pickle file /home/circleci/.conda/pkgs/cache/3e39a7aa.q
 2284.191     0.004 DEBUG conda.core.subdir_data:_load(210): Using cached repodata for https://repo.anaconda.com/pkgs/r/noarch at /home/circleci/.conda/pkgs/cache/4ea078d6.json. Timeout in 13 sec
 2284.192     0.004 DEBUG conda.core.subdir_data:_load(210): Using cached repodata for https://repo.anaconda.com/pkgs/free/noarch at /home/circleci/.conda/pkgs/cache/d77296e0.json. Timeout in 12 sec
 2284.192     0.000 DEBUG conda.core.subdir_data:_load(210): Using cached repodata for https://repo.anaconda.com/pkgs/r/linux-64 at /home/circleci/.conda/pkgs/cache/2ce54b42.json. Timeout in 12 sec
```

So I think either put it behind a different flag, or else we just have `CONDA_INSTRUMENTATION_ENABLED` off by default in master.

Reply:
Or maybe we could just attach it to the `conda.resolve` logger for now?

Reply:
And also maybe figure out some type of formatting that condenses the character count?

Reply:
> attach it to the `conda.resolve` logger for now

I think we can also use it for `conda.common.logic`, `conda.core.solve`, ...
And since it's not user-facing, i.e., nothing with a maintenance burden, I'm not that much opposed to adding another environment variable.

> formatting that condenses the character count

It is intentionally that we get a fixed width prefix here, to get a more uniform and less "jumpy" output here. But I guess the `9.3f` float formatting is a little overkill since it's a) a bit pessimistic as it dimensioned for days instead of minutes, and b) we only need to measure at most hundredths instead of thousandths of seconds. `7.2f` should be fine, i.e., being able to measure everything from 0.01 seconds to 2 3/4 hours.